File: client/ITI-graduation-project-Advilaw-Frontend-main/src/app/types/Lawyers/LawyerPaymentListDTO.ts
export interface LawyerPaymentListDTO {
  id: number;
  type: PaymentType;
  amount: number;
  senderId?: string;
  senderName: string;
  senderImgUrl: string;
  createdAt: string; // or Date if you're parsing it into Date objects
}

export enum PaymentType {
  SessionPayment = 0,
  RefundPayment,
  SubscriptionPayment,
  WithdrawPayment,
}

File: server/ITI-graduation-project-Advilaw-main/AdviLaw.Application/Features/PaymentSection/DTOs/LawyerPaymentListDTO.cs
using AdviLaw.Domain.Entites.PaymentSection;

namespace AdviLaw.Application.Features.PaymentSection.DTOs
{
    public class LawyerPaymentListDTO
    {
        public int Id { get; set; }
        public PaymentType Type { get; set; } = PaymentType.SessionPayment;
        public int Amount { get; set; } //mapped
        public string? SenderId { get; set; }
        public string SenderName { get; set; } = string.Empty; //mapped
        public string SenderImgUrl { get; set; } = string.Empty; //mapped
        public DateTime CreatedAt { get; set; }
    }
}
